### ISSUE TITLE
502 Gateway Error on Digital Ocean Droplets From 1-click install

### ISSUE THREAD

**Issue by beardedinbinary** [strapi/strapi#8586]
Existing strapi and new strapi one-click install droplets on digital ocean have a 502 gateway error. 

I created a Strapi site with the 1-click install on Digital Ocean, site had been running fine for about 2 weeks. This morning now the site has a 502 error. I've restored from various backups and had the same experience. 

Now I've created a new DO droplet with the 1-click strapi install to test if it was an issue isolated to my existing site. The newly created Strapi site is showing the same 502 error.

**Comment by derrickmehaffy** [comment]
Hi @beardedinbinary for new one-click instances it takes about 3 to 5 minutes for everything to setup.

On your existing instance can you please provide your pm2 logs, if you SSH as the Strapi user on your one-click instance the logs should be in `~/.pm2/logs`

**Comment by beardedinbinary** [comment]
@derrickmehaffy Thanks for your quick response. Here are the logs from the server that has been up since last week and is now experiencing the 502 gateway issue. The new install on another server has the same issue still. I have changed nothing in either environments. 

[strapi-development-error.log](https://github.com/strapi/strapi/files/5502462/strapi-development-error.log)
[strapi-development-out.log](https://github.com/strapi/strapi/files/5502463/strapi-development-out.log)

**Comment by derrickmehaffy** [comment]
`[2020-11-06T16:54:52.452Z] error error: relation "learn_articles_learn_categories__learn_categories_learn_article" already exists`

You have an issue in the database, did you recently make changes to the structure of some of your content-types like adding or removing anything? (Also for the time being we need to stop the server so it's not constantly trying to restart, use `pm2 stop strapi-development` to temporarily shut down the Strapi instance)

**Comment by derrickmehaffy** [comment]
As for the new instance, I just spun up a fresh test instance and it appears to be functioning fine? 

http://192.81.216.60/admin

**Comment by derrickmehaffy** [comment]
Yeah based on the logs:

```
[2020-11-06T00:07:47.164Z] debug GET /content-type-builder/components (117 ms) 200
[2020-11-06T00:07:47.183Z] debug GET /content-type-builder/content-types (125 ms) 200
[2020-11-06T00:07:47.196Z] debug GET /content-type-builder/reserved-names (131 ms) 200
[2020-11-06T00:14:22.537Z] info File changed: /srv/strapi/strapi-development/api/ingredient/models/ingredient.settings.json
[2020-11-06T00:14:22.539Z] debug PUT /content-type-builder/content-types/application::ingredient.ingredient (231 ms) 201
[2020-11-06T00:14:22.542Z] info The server is restarting

[2020-11-06T00:14:26.326Z] debug ⛔️ Server wasn't able to start properly.
[2020-11-06T00:14:26.328Z] error error: relation "learn_articles_learn_categories__learn_categories_learn_article" already exists
```

You modified the `/srv/strapi/strapi-development/api/ingredient/models/ingredient.settings.json` model via the content-type builder. Can you tell me what you changed?

**Comment by beardedinbinary** [comment]
@derrickmehaffy Ah, I'll shut down the strapi instance now. 

My client must have modified the content-type, instead of modifying a record. I'm not sure what changed.

**Comment by beardedinbinary** [comment]
@derrickmehaffy I tried running the PM2 command to shut it down and saw this 

`[PM2] Spawning PM2 daemon with pm2_home=/root/.pm2
[PM2] This PM2 is not UP TO DATE
[PM2] Upgrade to version 4.5.0
[PM2] PM2 Successfully daemonized
[PM2][ERROR] Process or Namespace strapi-development not found`

**Comment by beardedinbinary** [comment]
@derrickmehaffy looking at the model file, it looks like this was added to the model before we had this issue. 

```"DropDownPriority": {
      "type": "enumeration",
      "enum": [
        "one",
        "two"
      ]
    }```

**Comment by derrickmehaffy** [comment]
Run a `pm2 list` and it will give you the name or id. You can use `pm2 stop 0` or whatever the ID is or `pm2 stop some-name`

To me it looks like some kind of relation or component was added but there is a conflict in the database, likely from it being added, deleted, then readded with different information.

To quickly fix you delete to delete that modification, longer fix is to use the pg-cli to delete the problem relational column/table in the database. I would need to see the model structure to know for sure.

As a side point if you are using the one-click in production, that is strongly advised against and we state in our one-click docs it should only be used for testing right now.

**Comment by derrickmehaffy** [comment]
(That enum field shouldn't be causing the problem, as the name provided in those logs would be a relation or a component that was modified/added) 

Can you give me the `/srv/strapi/strapi-development/api/ingredient/models/ingredient.settings.json` file?

**Comment by beardedinbinary** [comment]
@derrickmehaffy Got it. Here's the settings file. Would it be possible to just restore a previous version of that file on the server, start up pm2 again, and have it work? Thanks again!
[ingredient.settings.json.txt](https://github.com/strapi/strapi/files/5502644/ingredient.settings.json.txt)

**Comment by derrickmehaffy** [comment]
> Would it be possible to just restore a previous version of that file on the server, start up pm2 again, and have it work?

Unless you have the project being commited into Git or some kind of source control there won't be a history, this is the primary reason we don't suggest the one-click for production usage.

Let me take a look at the file

**Comment by derrickmehaffy** [comment]
Looks like everything is fine there :thinking: Do me a favor and shoot me an email at support@strapi.io so we can do some troubleshooting outside the normal public eye please :)

**Comment by derrickmehaffy** [comment]
Hi @beardedinbinary so as discussed the issue related to a relation that was constructed between two models and it appears the issue was the relation was deleted from the model structures but not in the database (Related GitHub issue: https://github.com/strapi/strapi/issues/1114) and was recreated but the existing table in the database wasn't removed.

In a short summery of what we did to fix it was to stop the running server, temporarily remove the relation on both sides from the Strapi model settings files, then clean up the problem database table from your PG database.

From there we restarted the server and recreated the relation via the CTB. In your case there was no important information in the table so we did not need to back it up.

As mentioned previously going forward it's best to use our Forum for environmental issues but it's certainly understandable that this issue looked like a bug to you. I have gone ahead and deleted the messages where you provided some debug logs and model files just to ensure you didn't have any private data that leaks.
